static/app/utils/profiling/renderers/textRenderer.tsx

```diff
@@ -126,19 +131,53 @@ class TextRenderer {
       // Offset x by 1x the padding
       const x = frameInPhysicalSpace[0] + HALF_SIDE_PADDING;
 
-      this.context.fillText(
-        trimTextCenter(
-          frame.frame.name,
-          findRangeBinarySearch(
-            {low: 0, high: paddedRectangleWidth},
-            n => this.measureAndCacheText(frame.frame.name.substring(0, n)),
-            paddedRectangleWidth
-          )[0]
-        ),
-        x,
-        y
+      const frameName = frame.frame.name;
+      const trim = trimTextCenter(
+        frameName,
+        findRangeBinarySearch(
+          {low: 0, high: paddedRectangleWidth},
+          n => this.measureAndCacheText(frame.frame.name.substring(0, n)).width,
+          paddedRectangleWidth
+        )[0]
       );
 
+      const {text: trimText} = trim;
+
+      if (flamegraphSearchResults) {
+        const frameId = getFlamegraphFrameSearchId(frame);
+        const frameSearchResult = flamegraphSearchResults[frameId];
+
+        if (frameSearchResult) {
+          const highlightFillStyle = `rgb(${this.theme.COLORS.HIGHLIGHTED_LABEL_COLOR.join(
+            ', '
+          )})`;
+
+          for (const matchIndices of frameSearchResult.matchIndices) {
+            const highlightedBounds = computeHighlightedBounds(matchIndices, trim);
+            if (!highlightedBounds) {
+              continue;
+            }
+            const [startIndex, endIndex] = highlightedBounds;
+            const highlightTextSize = this.measureAndCacheText(
+              trimText.substring(startIndex, endIndex)
+            );
+            const frontMatter = trimText.slice(0, startIndex);
+            const startHighlightX = this.measureAndCacheText(frontMatter).width;
+            this.context.fillStyle = highlightFillStyle;
+            this.context.fillRect(
+              startHighlightX + x,
+              y - highlightTextSize.fontBoundingBoxAscent,
+              highlightTextSize.width,
+              highlightTextSize.fontBoundingBoxAscent +
+                highlightTextSize.fontBoundingBoxDescent
+            );
+          }
+        }
+      }
+
+      this.context.fillStyle = this.theme.COLORS.LABEL_FONT_COLOR;
```

Comment:
nit: at the top of this `draw` method, the fill style is being unnecessarily set now that we set it here, let's remove that